### PR TITLE
fix: Duplicates in dropdown from different data sources

### PR DESCRIPTION
**Pull request by alainajaffer**

### CODE REVIEW

File: site/components/ViewAllContents.tsx [hunk 1]
```diff
@@ -476,6 +476,7 @@ const ViewAllContents = ({
 }: ViewAllContentProps): ReactElement => {
     const [selectedServices, setSelectedServices] = useState<Service[]>([]);
     const [selectedOperators, setSelectedOperators] = useState<ConsequenceOperators[]>([]);
+    const [servicesDataSource, setServicesDataSource] = useState<Datasource | undefined>();
```
Comment (ljones92) [inline site/components/ViewAllContents.tsx:479]:
```suggestion
    const [servicesDataSource, setServicesDataSource] = useState<Datasource | "all">("all");
```

If the dropdown is setting this to a value of "all" then we should ideally include it in the type here and default to it

File: site/components/ViewAllContents.tsx [hunk 2]
```diff
@@ -698,10 +700,22 @@ const ViewAllContents = ({
             ...filterServices(servicesTndsData),
         ]);
 
+        setCombinedServicesList(combinedServices);
+
         setOperatorsList(operators);
         setServicesList(combinedServices);
     };
 
+    useEffect(() => {
```
Comment (ljones92) [inline site/components/ViewAllContents.tsx:709]: Then we can streamline this useEffect to be something like:

```ts
setServicesList(
    servicesDataSource === "all"
        ? combinedServicesList
        : combinedServicesList.filter((service) => service.dataSource === servicesDataSource),
);
```

File: site/components/search/ServiceSearch.tsx [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
 import Select, { ControlProps, GroupBase, OptionProps } from "react-select";
 import { getServiceLabel } from "../../utils";
+import Dropdown from "../form/Select";
 
 interface ServiceSearchProps {
     services: Service[];
     setSelectedServices: Dispatch<SetStateAction<Service[]>>;
     selectedServices: Service[];
     reset?: boolean;
+    showToggle?: boolean;
```
Comment (ljones92) [inline site/components/search/ServiceSearch.tsx:13]: Do we need this toggle? I think this is the only place this component is used

File: site/components/search/ServiceSearch.tsx [hunk 4]
```diff
@@ -1,20 +1,28 @@
...
+    dataSource?: string;
```
Comment (ljones92) [inline site/components/search/ServiceSearch.tsx:14]: And then this and `handleDataSourceUpdate` wouldn't need to be optional either

File: site/components/search/ServiceSearch.tsx [hunk 5]
```diff
@@ -40,42 +48,62 @@ const ServiceSearch = ({
...
+                    }}
+                    id="service-filter"
+                    instanceId="service-filter-dropdown"
+                    inputId="service-filter-dropdown-value"
+                    menuPlacement="auto"
+                    menuPosition="fixed"
+                    value={null}
+                />
+            </div>
+            {showToggle && handleDataSourceUpdate && (
```
Comment (ljones92) [inline site/components/search/ServiceSearch.tsx:89]: And then we could get rid of this check

File: site/components/search/ServiceSearch.tsx (hunk 4 above)
Comment (alainajaffer) [reply site/components/search/ServiceSearch.tsx]: done

File: site/components/search/ServiceSearch.tsx (hunk 5 above)
Comment (alainajaffer) [reply site/components/search/ServiceSearch.tsx]: done

File: site/components/search/ServiceSearch.tsx (hunk 3 above)
Comment (alainajaffer) [reply site/components/search/ServiceSearch.tsx]: done

File: site/components/ViewAllContents.tsx (hunk 2 above)
Comment (alainajaffer) [reply site/components/ViewAllContents.tsx:709]: done

File: site/components/ViewAllContents.tsx (hunk 1 above)
Comment (alainajaffer) [reply site/components/ViewAllContents.tsx]: done

File: site/components/search/ServiceSearch.tsx [hunk 6]
```diff
@@ -40,42 +48,62 @@ const ServiceSearch = ({
...
+                <div className="w-1/3">
+                    <Dropdown
+                        inputName="servicesDataSource"
+                        display="Services data source"
+                        value={dataSource || "all"}
+                        defaultDisplay="Select a services data source"
+                        selectValues={[
+                            { display: "All sources", value: "all" },
+                            { display: "BODs", value: Datasource.bods },
+                            { display: "TNDs", value: Datasource.tnds },
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/components/search/ServiceSearch.tsx]: done

File: site/components/search/ServiceSearch.tsx [hunk 7]
```diff
@@ -40,42 +48,62 @@ const ServiceSearch = ({
...
+            {showToggle && handleDataSourceUpdate && (
+                <div className="w-1/3">
+                    <Dropdown
+                        inputName="servicesDataSource"
+                        display="Services data source"
+                        value={dataSource || "all"}
+                        defaultDisplay="Select a services data source"
+                        selectValues={[
+                            { display: "All sources", value: "all" },
+                            { display: "BODs", value: Datasource.bods },
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/components/search/ServiceSearch.tsx]: done

**Review by ljones92 — APPROVED** [review]

**Pull request merged**